File: tiq-e/strings/1/main.ts
import {performance} from 'perf_hooks';

/**
 Do not return anything, modify s in-place instead.
 */
function reverseString(s: string[]): void {
  const [half, max] = [s.length >> 1, s.length - 1]
  for (let i = 0; i < half; i++) {
    [s[i], s[max - i]] = [s[max - i], s[i]]
  }
}

let arr = ['h', 'e', 'l', 'l', 'o']

let start = performance.now();
reverseString(arr);
let end = performance.now();
console.log('Duration:', end - start);
console.log('Result:', arr)

File: tiq-e/strings/1/main.py
from time import perf_counter
from typing import List


class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        half, max = len(s) >> 1, len(s) - 1
        for i in range(half):
            s[i], s[max - i] = s[max - i], s[i]


arr = ['h', 'e', 'l', 'l', 'o']

start = perf_counter()
Solution().reverseString(arr)
end = perf_counter()
print("Duration: " + str(end - start))
print("Result: " + str(arr))
